lib/content_change_email_status_count.rb

```diff
@@ -0,0 +1,47 @@
+class ContentChangeEmailStatusCount
+  def initialize(content_change)
+    @content_change = content_change
+    @email_statuses = email_statuses
+    @email_status_counts = email_status_counts
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    puts <<~TEXT
+      -------------------------------------------
+      Email status counts for Content Change #{@content_change.id}
+      -------------------------------------------
+
+      Sent emails: #{@email_status_counts[:sent]}
+
+      Pending emails: #{@email_status_counts[:pending]}
+
+      Failed emails: #{@email_status_counts[:failed]}
+
+      -------------------------------------------
+    TEXT
+  end
+
+private
+
+  def email_statuses
+    @content_change.subscription_contents.map do |sub_contents|
```

Comment:
I suspect this will lead to many queries on certain content changes, perhaps we could do it in stages, something like below to save on the number of queries:

```ruby
subscription_contents_ids = @content_change.subscription_contents.pluck(:id)
email_ids = SubscriptionContent.where(id: subscription_contents_ids).pluck(:email_id)
Email.where(id: email_ids).pluck(:status)
```

We should have indexes on all those ID fields so it should be reasonably fast.

Reply:
In fact, just thinking you could even do the count query this way:

```
Email.where(id: email_ids).group(:status).count
```

Reply:
Thanks @thomasleese, have updated the queries